### ISSUE TITLE
'\n' in method default args gets munged in autodoc output

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1502]
If I have a method with a "\n" as a default argument value, then in the autodoc output, this gets transformed to a "n". This is the same issue as #376 but now for methods.

The follow patch resolves this:

```
#!diff

diff -r 0e789eafb3bb sphinx/ext/autodoc.py
--- a/sphinx/ext/autodoc.py     Thu Jun 19 12:14:47 2014 -0400
+++ b/sphinx/ext/autodoc.py     Tue Jul 01 12:18:09 2014 +0200
@@ -1258,7 +1258,10 @@
         argspec = getargspec(self.object)
         if argspec[0] and argspec[0][0] in ('cls', 'self'):
             del argspec[0][0]
-        return inspect.formatargspec(*argspec)
+        args = inspect.formatargspec(*argspec)
+        # escape backslashes for reST
+        args = args.replace('\\', '\\\\')
+        return args

     def document_members(self, all_members=False):
         pass
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1502
- Originally reported by: Geert Jansen
- Originally created at: 2014-07-01T12:19:36.549

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-07-01 15:09:54+00:00_
- In autodoc, fix display of parameter defaults containing backslashes. Closes #1502

→ <<cset b3aa4aa81bb7311136587c134f805b86d2435fb3>>

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-07-01 15:10:04+00:00_

Thanks!
